### ISSUE TITLE
Visualize or get classes from the ClassificationTree explain_global() leaves

### ISSUE THREAD

**Issue by federicocau** [interpretml/interpret#569]
Hello there, 

I was wondering if there is a way to visualize the Classification Tree classes (leaves) in the explain_global() plot, or even better, print/return them somehow. I was inspecting the documentation but found nothing about getting the leaves' classes (the plot only shows observation and impurity).

Thank you!

**Comment by paulbkoch** [comment]
(answered by claude code)

Hi @federicocau — the underlying sklearn tree is accessible, so you can extract this information today.

Getting leaf classes and decision rules:

from sklearn.tree import export_text

# Access the underlying sklearn DecisionTreeClassifier
tree = clf._model()

# Print rules with class labels
print(export_text(tree, feature_names=clf.feature_names_in_))

For class counts at each node, the raw data is in tree.tree_.value, and clf.classes_ maps indices to class labels.

Getting precision/recall per rule isn't built in, but you can extract leaf assignments and compute them:

leaf_ids = tree.apply(X_test)
for leaf in np.unique(leaf_ids):
    mask = leaf_ids == leaf
    # compute precision/recall for samples in this leaf

Regarding the visualization: you're right that explain_global() only shows observation counts and impurity in leaf labels, not the predicted class.
